flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonRowDataSerDeSchemaTest.java

```diff
@@ -326,22 +322,30 @@ public void testDeserializationMissingNode() throws Exception {
 		deserializationSchema = deserializationSchema = new JsonRowDataDeserializationSchema(
 			schema, WrapperTypeInfo.of(schema), true, false, TimestampFormat.ISO_8601);
 
-		thrown.expect(IOException.class);
-		thrown.expectMessage("Failed to deserialize JSON '{\"id\":123123123}'");
-		deserializationSchema.deserialize(serializedJson);
+		String errorMessage = "Failed to deserialize JSON '{\"id\":123123123}'.";
+		try {
+			deserializationSchema.deserialize(serializedJson);
+			throw new RuntimeException("expecting exception message: " + errorMessage);
+		} catch (Throwable t) {
+			assertEquals(errorMessage, t.getMessage());
+		}
 
 		// ignore on parse error
 		deserializationSchema = new JsonRowDataDeserializationSchema(
 			schema, WrapperTypeInfo.of(schema), false, true, TimestampFormat.ISO_8601);
 		actual = convertToExternal(deserializationSchema.deserialize(serializedJson), dataType);
 		assertEquals(expected, actual);
 
-		thrown.expect(IllegalArgumentException.class);
-		thrown.expectMessage("JSON format doesn't support failOnMissingField and ignoreParseErrors are both enabled");
-		// failOnMissingField and ignoreParseErrors both enabled
-		//noinspection ConstantConditions
-		new JsonRowDataDeserializationSchema(
-			schema, WrapperTypeInfo.of(schema), true, true, TimestampFormat.ISO_8601);
+		errorMessage = "JSON format doesn't support failOnMissingField and ignoreParseErrors are both enabled.";
+		try {
+			// failOnMissingField and ignoreParseErrors both enabled
+			//noinspection ConstantConditions
+			new JsonRowDataDeserializationSchema(
+				schema, new RowDataTypeInfo(schema), true, true, TimestampFormat.ISO_8601);
```

Comment:
Use `WrapperTypeInfo.of(schema)` instead of `new RowDataTypeInfo(schema)`. We are in the progress of removing `RowDataTypeInfo`.